presto-main/src/main/java/com/facebook/presto/transaction/InMemoryTransactionManager.java

```diff
@@ -568,11 +568,13 @@ public synchronized void checkConnectorWrite(ConnectorId connectorId)
 
             ConnectorTransactionMetadata writeConnector = connectorIdToMetadata.get(writeConnectorId);
             Supplier<ListenableFuture> commitFunctionNamespaceTransactions = () -> functionNamespaceFuture;
-            ListenableFuture<?> commitFuture = Futures.transformAsync(finishingExecutor.submit(writeConnector::commit), ignored -> commitFunctionNamespaceTransactions.get(), directExecutor());
-            ListenableFuture<?> readOnlyCommitFuture = Futures.transformAsync(commitFuture, ignored -> commitReadOnlyConnectors.get(), directExecutor());
-            addExceptionCallback(readOnlyCommitFuture, this::abortInternal);
+            ListenableFuture<?> readOnlyCommitFuture = Futures.transformAsync(
+                    commitFunctionNamespaceTransactions.get(), ignored -> commitReadOnlyConnectors.get(), directExecutor());
+            ListenableFuture<?> writeCommitFuture = Futures.transformAsync(
+                    readOnlyCommitFuture, ignored -> finishingExecutor.submit(writeConnector::commit), directExecutor());
+            addExceptionCallback(writeCommitFuture, this::abortInternal);
```

Comment:
looks like the commit order is changed here. can you briefly explain this change.

Reply:
Previously, the commit result is not returned; therefore, the order of the commits are not important. Here we need to return the commit result for further handling, the order matters. Two cases: for read query commit, the order is the same as previous. For write query commits, write commits should be the last one in order to return the result (e.g., create a table, or add partitions). Previously the write commit results are swallowed before the read commits.

Reply:
Can we please add a comment explaining the logic you mentioned above.

Reply:
Can we please add a comment explaining the logic you mentioned above.

Reply:
I have added a simple version. Do you think if it is enough?